### PR TITLE
Improve BBEs

### PR DESCRIPTION
**Pull request by pubudu91**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [x] Ballerina By Examples

### CODE REVIEW

File: examples/objects/objects.description [hunk 1]
```diff
@@ -1,2 +1,3 @@
-// In Ballerina, objects are collections of public/private/protected typed fields along with attached
-// functions that allow you to create new, user-defined data types with behavior.
+// Objects are a behaviourial type. An object is a combination of public and private fields along with a
+// set of associated functions (called methods) that can be used to manipulate the fields. that allow you
```
Comment (MaryamZi) [inline examples/objects/objects.description]:
```suggestion
// set of associated functions (called methods) that can be used to manipulate the fields that allow you
```

File: examples/records/records.bal [hunk 2]
```diff
@@ -1,12 +1,17 @@
 import ballerina/io;
 
+// This defines an open record type named `Student` which in addition to the fields defined here, allows additional
+// string fields.
```
Comment (MaryamZi) [inline examples/records/records.bal]:
```suggestion
// `string` fields.
```

File: examples/abstract-objects/abstract_objects.bal [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
-    // Non-abstract object can have a constructor method.
+    // Non-abstract objects can have initializers.
     function __init(int age, string firstName, string lastName) {
         self.age = age;
         self.firstName = firstName;
         self.lastName = lastName;
     }
 
-    // Member function should have a body.
+    // Methods should have a body.
```
[the remark this answers is not in the document]
Comment (pubudu91) [reply examples/abstract-objects/abstract_objects.bal]:
```suggestion
    // Methods should have a definition either within the object or outside of the object.
```
How about this?

File: examples/object-type-reference/object_type_reference.bal [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
     // have a body.
     function getFullName() returns string;
 
 };
 
-// Define another `abstract object` called `Employee`, which references the `Person` object.
+// Define another abstract object called `Employee`, which references the `Person` object.
 type Employee abstract object {
-    // Add a reference to the `Person`. Note that only abstract objects can be referred.
+    // Add a reference to the `Person` object type. Note that only abstract objects can be referred to.
```
[the remark this answers is not in the document]
Comment (pubudu91) [reply examples/object-type-reference/object_type_reference.bal]: Hm yeah, makes sense. Can you add a suggestion?

[the remark this answers is not in the document]
Comment (pubudu91) [reply examples/object-type-reference/object_type_reference.bal]:
```suggestion
    // Add a reference to the `Person` object type. Note that only abstract objects can be referred.
```

File: examples/abstract-objects/abstract_objects.bal (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaryamZi) [reply examples/abstract-objects/abstract_objects.bal]: +1, but body is also OK right? Might sound more familiar.

File: examples/objects/objects.bal [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import ballerina/io;
 
-// Defines an `object` called `Person`. It contains `public`, `private` and protected fields along with their types.
-// For fields, the default value must be defined.
+// Defines an object called `Person`. It contains `public`, `private` and protected fields along with their types.
```
[the remark this answers is not in the document]
Comment (pubudu91) [reply examples/objects/objects.bal]: Yeah.. Wasn't sure of a term for this.. so just left it at it is..

[the remark this answers is not in the document]
Comment (pubudu91) [reply examples/objects/objects.bal]: The spec just says "public and private fields" though

**Comment by praneesha** [comment]
Issue #14969 created for the doc review of this.

**Pull request merged**